server/src/handlers/chunk_handler.rs

```diff
@@ -2674,3 +2675,73 @@ pub fn check_completion_param_validity(
 
     Ok(())
 }
+
+#[derive(Serialize, Deserialize, Debug, ToSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum CrawlInterval {
+    Daily,
+    Weekly,
+    Monthly,
+}
+
+#[derive(Serialize, Deserialize, Debug, ToSchema)]
+pub struct CrawlRequestBody {
+    pub site: String,
+    pub interval: Option<CrawlInterval>,
+}
+
+#[derive(Serialize, Deserialize, Debug, ToSchema)]
+pub struct CrawlResponse {
+    pub scrape_id: uuid::Uuid,
+}
+
+/// Crawl a Website
+///
+/// Crawl a website use firecrawl and make the data available for search.
+#[utoipa::path(
+    post,
+    path = "/chunk/crawl",
+    context_path = "/api",
+    tag = "Chunk",
+    request_body(content = CrawlRequestBody, description = "JSON request payload to crawl a site", content_type = "application/json"),
+    responses(
+        (status = 200, description = "The crawl id for the website", body = CrawlResponse),
+        (status = 400, description = "Service error relating to crawling", body = ErrorResponseBody),
+    ),
+    params(
+        ("TR-Dataset" = String, Header, description = "The dataset id or tracking_id to use for the request. We assume you intend to use an id if the value is a valid uuid."),
+    ),
+    security(
+        ("ApiKey" = ["readonly"]),
+    )
+)]
+pub async fn crawl(
+    data: web::Json<CrawlRequestBody>,
+    pool: web::Data<Pool>,
+    redis_pool: web::Data<RedisPool>,
+    _user: LoggedUser,
+    dataset_org_plan_sub: DatasetAndOrgWithSubAndPlan,
+) -> Result<HttpResponse, actix_web::Error> {
+    let data = data.into_inner();
+    let scrape_id = crawl_site(data.site.clone())
+        .await
+        .map_err(|err| ServiceError::BadRequest(format!("Could not crawl site: {}", err)))?;
+
+    let interval = match data.interval {
+        Some(CrawlInterval::Daily) => std::time::Duration::from_secs(60 * 60 * 24),
+        Some(CrawlInterval::Weekly) => std::time::Duration::from_secs(60 * 60 * 24 * 7),
+        Some(CrawlInterval::Monthly) => std::time::Duration::from_secs(60 * 60 * 24 * 30),
+        None => std::time::Duration::from_secs(0),
+    };
+
+    let scrape_id = create_crawl_request(
+        data.site,
+        dataset_org_plan_sub.dataset.id,
+        scrape_id,
+        interval,
+        pool,
+        redis_pool,
+    )
+    .await?;
+    Ok(HttpResponse::Ok().json(CrawlResponse { scrape_id }))
+}
```

Comment:
We should move this to something that happens on dataset creation and update instead of having it be its own route.

I tend to think this is something people will almost exclusively do via the dashboard UI.